install/kubernetes/cilium/templates/cilium-ca-secret.yaml

```diff
@@ -11,6 +11,12 @@ kind: Secret
 metadata:
   name: {{ .commonCASecretName }}
   namespace: {{ .Release.Namespace }}
+  annotations:
+    # If the tls.auto.method is changed from "helm" to "cronJob" then we want
+    # to keep this CA, so that it can be used by certgen's --ca-reuse-secret
+    # flag. Keeping this CA secret also required in the case that it has been
+    # manually copied from another cluster between runs of Helm.
+    "helm.sh/resource-policy": keep
```

Comment:
How will this interact with a `helm uninstall` followed by a `helm install`? According to [the docs](https://helm.sh/docs/howto/charts_tips_and_tricks/#tell-helm-not-to-uninstall-a-resource) this will break helm install. Can we somehow tell Helm to not freak out if this resource exists in `helm install`?

(I think this might also be flagged in CI, as we had problems with certificates not being removed between CI runs, but you would have to run `/test` to be sure if this is still an issue)

Reply:
I think it will only break `helm install --replace`, which is not recommended for production IIUC :thinking:

Reply:
So `helm install` (without `--replace`) will not complain about the resource already existing?

Reply:
Here's a link to the issue we had previously: https://github.com/cilium/cilium/pull/17591

Reply:
In this case, helm can adopt the existing resource if it's annotated/labeled correctly. I can't find the official docs on it, but it's an official feature of helm v3. https://stackoverflow.com/questions/61177820/use-helm-3-for-existing-resources-deployed-with-kubectl

Reply:
Thanks for digging this up Chance, in that case this looks good to me!